ion/reader.go

```diff
@@ -98,6 +98,9 @@ type Reader interface {
 	// It returns nil if there is no current value or the current value has no annotations.
 	Annotations() []string
```

Comment:
I was under the impression that we'd be replacing this method (`Annotations() []string`) instead of adding a new one alongside it since `[]string` makes error reporting challenging.